Fails to build assets on Node v7.2.1

### Expected behavior

It builds.

### Actual behavior

It certainly does not:

```

> talk@0.0.1 build /Users/wyatt/Code/github.com/coralproject/talk
> NODE_ENV=production ./node_modules/.bin/webpack --config webpack.config.js --bail

node_modules/nan
resolve failed:  { Error: Cannot find module 'eslint-module-utils'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.requireRelative.resolve (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/require-relative/index.js:30:17)
    at resolve (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:26:26)
    at findAndRemove (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:83:11)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:126:13
    at Array.map (native)
    at loadPreset (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:118:29)
    at module.exports (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/index.js:95:19)
    at Object.<anonymous> (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-preset-es2015-minimal/index.js:5:18)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:302:17
    at Array.map (native)
    at OptionManager.resolvePresets (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:270:20)
    at OptionManager.mergePresets (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:259:10)
    at OptionManager.mergeOptions (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:244:14)
    at OptionManager.init (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:374:12)
    at File.initOptions (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/index.js:216:65)
    at new File (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/index.js:139:24)
    at Pipeline.transform (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/pipeline.js:46:16)
    at transpile (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-loader/lib/index.js:41:20)
    at Object.module.exports (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-loader/lib/index.js:138:12)
    at WEBPACK_CORE_LOADER_EXECUTION (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:155:71)
    at runSyncOrAsync (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:155:93)
    at nextLoader (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:290:3)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:259:5
    at Storage.finished (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:38:16)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/graceful-fs/graceful-fs.js:78:16
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:415:3) code: 'MODULE_NOT_FOUND' } eslint-module-utils
resolve failed:  { Error: Cannot find module 'caniuse-db'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.requireRelative.resolve (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/require-relative/index.js:30:17)
    at resolve (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:26:26)
    at findAndRemove (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:83:11)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:126:13
    at Array.map (native)
    at loadPreset (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:118:29)
    at module.exports (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/index.js:95:19)
    at Object.<anonymous> (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-preset-es2015-minimal/index.js:5:18)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:302:17
    at Array.map (native)
    at OptionManager.resolvePresets (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:270:20)
    at OptionManager.mergePresets (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:259:10)
    at OptionManager.mergeOptions (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:244:14)
    at OptionManager.init (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:374:12)
    at File.initOptions (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/index.js:216:65)
    at new File (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/index.js:139:24)
    at Pipeline.transform (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/pipeline.js:46:16)
    at transpile (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-loader/lib/index.js:41:20)
    at Object.module.exports (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-loader/lib/index.js:138:12)
    at WEBPACK_CORE_LOADER_EXECUTION (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:155:71)
    at runSyncOrAsync (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:155:93)
    at nextLoader (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:290:3)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:259:5
    at Storage.finished (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:38:16)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/graceful-fs/graceful-fs.js:78:16
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:415:3) code: 'MODULE_NOT_FOUND' } caniuse-db
resolve failed:  { Error: Cannot find module 'babel-runtime'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.requireRelative.resolve (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/require-relative/index.js:30:17)
    at resolve (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:26:26)
    at findAndRemove (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:83:11)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:126:13
    at Array.map (native)
    at loadPreset (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:118:29)
    at module.exports (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/index.js:95:19)
    at Object.<anonymous> (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-preset-es2015-minimal/index.js:5:18)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:302:17
    at Array.map (native)
    at OptionManager.resolvePresets (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:270:20)
    at OptionManager.mergePresets (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:259:10)
    at OptionManager.mergeOptions (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:244:14)
    at OptionManager.init (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:374:12)
    at File.initOptions (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/index.js:216:65)
    at new File (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/index.js:139:24)
    at Pipeline.transform (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/pipeline.js:46:16)
    at transpile (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-loader/lib/index.js:41:20)
    at Object.module.exports (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-loader/lib/index.js:138:12)
    at WEBPACK_CORE_LOADER_EXECUTION (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:155:71)
    at runSyncOrAsync (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:155:93)
    at nextLoader (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:290:3)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:259:5
    at Storage.finished (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:38:16)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/graceful-fs/graceful-fs.js:78:16
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:415:3) code: 'MODULE_NOT_FOUND' } babel-runtime
resolve failed:  { Error: Cannot find module 'webpack-core'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.requireRelative.resolve (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/require-relative/index.js:30:17)
    at resolve (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:26:26)
    at findAndRemove (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:83:11)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:126:13
    at Array.map (native)
    at loadPreset (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:118:29)
    at module.exports (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/index.js:95:19)
    at Object.<anonymous> (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-preset-es2015-minimal/index.js:5:18)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:302:17
    at Array.map (native)
    at OptionManager.resolvePresets (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:270:20)
    at OptionManager.mergePresets (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:259:10)
    at OptionManager.mergeOptions (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:244:14)
    at OptionManager.init (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:374:12)
    at File.initOptions (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/index.js:216:65)
    at new File (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/index.js:139:24)
    at Pipeline.transform (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/pipeline.js:46:16)
    at transpile (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-loader/lib/index.js:41:20)
    at Object.module.exports (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-loader/lib/index.js:138:12)
    at WEBPACK_CORE_LOADER_EXECUTION (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:155:71)
    at runSyncOrAsync (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:155:93)
    at nextLoader (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:290:3)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:259:5
    at Storage.finished (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:38:16)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/graceful-fs/graceful-fs.js:78:16
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:415:3) code: 'MODULE_NOT_FOUND' } webpack-core
resolve failed:  { Error: Cannot find module '@kadira'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.requireRelative.resolve (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/require-relative/index.js:30:17)
    at resolve (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:26:26)
    at findAndRemove (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:83:11)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:126:13
    at Array.map (native)
    at loadPreset (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/lib/serialize.js:118:29)
    at module.exports (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/modify-babel-preset/index.js:95:19)
    at Object.<anonymous> (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-preset-es2015-minimal/index.js:5:18)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:302:17
    at Array.map (native)
    at OptionManager.resolvePresets (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:270:20)
    at OptionManager.mergePresets (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:259:10)
    at OptionManager.mergeOptions (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:244:14)
    at OptionManager.init (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/options/option-manager.js:374:12)
    at File.initOptions (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/index.js:216:65)
    at new File (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/file/index.js:139:24)
    at Pipeline.transform (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-core/lib/transformation/pipeline.js:46:16)
    at transpile (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-loader/lib/index.js:41:20)
    at Object.module.exports (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/babel-loader/lib/index.js:138:12)
    at WEBPACK_CORE_LOADER_EXECUTION (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:155:71)
    at runSyncOrAsync (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:155:93)
    at nextLoader (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:290:3)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/webpack-core/lib/NormalModuleMixin.js:259:5
    at Storage.finished (/Users/wyatt/Code/github.com/coralproject/talk/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:38:16)
    at /Users/wyatt/Code/github.com/coralproject/talk/node_modules/graceful-fs/graceful-fs.js:78:16
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:415:3) code: 'MODULE_NOT_FOUND' } @kadira
INVALID COMMIT MSG: does not match "<type>(<scope>): <subject>" !
undefined

npm ERR! Darwin 16.1.0
npm ERR! argv "/usr/local/Cellar/node/7.2.1/bin/node" "/usr/local/bin/npm" "run" "build"
npm ERR! node v7.2.1
npm ERR! npm  v3.10.9
npm ERR! code ELIFECYCLE
npm ERR! talk@0.0.1 build: `NODE_ENV=production ./node_modules/.bin/webpack --config webpack.config.js --bail`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the talk@0.0.1 build script 'NODE_ENV=production ./node_modules/.bin/webpack --config webpack.config.js --bail'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the talk package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     NODE_ENV=production ./node_modules/.bin/webpack --config webpack.config.js --bail
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs talk
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls talk
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/wyatt/Code/github.com/coralproject/talk/npm-debug.log
```

### Steps to reproduce behavior

Run in a docker container with:

```bash
# remove old node modules folder
rm -rf node_modules

# run in node docker container
docker run -ti --rm -v $PWD:/usr/local/app -w /usr/local/app node:7 bash

# verify node version
node --version

# install deps
npm install

# build assets
npm run build
```

Then you should see an error!

--- a/.babelrc
+++ b/.babelrc
@@ -2,7 +2,7 @@
   "sourceMaps": true,
   "presets": [
     "stage-0",
-    "es2015-minimal"
+    "es2015"
   ],
   "plugins": [
     ["transform-decorators-legacy"],

--- a/models/setting.js
+++ b/models/setting.js
@@ -142,11 +142,9 @@ SettingService.update = (settings) => Setting.findOneAndUpdate(selector, {
 }).then((settings) => {
 
   // Invalidate the settings cache.
-  return Promise.all([
-    cache.set('settings', settings, EXPIRY_TIME),
-    cache.invalidate('publicSettings')
-  ])
-  .then(() => settings);
+  return cache
+    .set('settings', settings, EXPIRY_TIME)
+    .then(() => settings);
 });
 
 /**

--- a/models/setting.js
+++ b/models/setting.js
@@ -84,7 +84,15 @@ SettingSchema.method('merge', function(src) {
  */
 SettingSchema.method('filterForUser', function(user = false) {
   if (!user || !user.roles.includes('admin')) {
-    return _.pick(this.toJSON(), ['moderation', 'infoBoxEnable', 'infoBoxContent']);
+    return _.pick(this.toJSON(), [
+      'moderation',
+      'infoBoxEnable',
+      'infoBoxContent',
+      'closeTimeout',
+      'closedMessage',
+      'charCountEnable',
+      'charCount'
+    ]);
   }
 
   return this.toJSON();
@@ -107,11 +115,6 @@ const SettingService = module.exports = {};
 const selector = {id: '1'};
 
 /**
- * The list of settings that can be viewed publicly.
- */
-const publicSettings = 'moderation infoBoxEnable infoBoxContent closeTimeout closedMessage charCountEnable charCount';
-
-/**
  * Cache expiry time in seconds for when the cached entry of the settings object
  * expires. 2 minutes.
  */
@@ -123,14 +126,6 @@ const EXPIRY_TIME = 60 * 2;
  */
 SettingService.retrieve = () => cache.wrap('settings', EXPIRY_TIME, () => {
   return Setting.findOne(selector);
-}).then((setting) => new Setting(setting));
-
-/**
- * Gets publicly available settings records
- * @return {Promise} settings the publicly viewable settings record
- */
-SettingService.public = () => cache.wrap('publicSettings', EXPIRY_TIME, () => {
-  return Setting.findOne(selector, publicSettings);
 }).then((setting) => new Setting(setting));
 
 /**

--- a/package.json
+++ b/package.json
@@ -86,7 +86,6 @@
     "babel-plugin-transform-react-jsx": "^6.8.0",
     "babel-polyfill": "^6.16.0",
     "babel-preset-es2015": "^6.18.0",
-    "babel-preset-es2015-minimal": "^2.1.0",
     "babel-preset-stage-0": "^6.16.0",
     "chai": "^3.5.0",
     "chai-http": "^3.0.0",

--- a/routes/api/stream/index.js
+++ b/routes/api/stream/index.js
@@ -38,7 +38,7 @@ router.get('/', (req, res, next) => {
       }),
 
     // Get the moderation setting from the settings.
-    Setting.public()
+    Setting.retrieve()
   ])
   .then(([asset, settings]) => {
 

--- a/tests/models/asset.js
+++ b/tests/models/asset.js
@@ -24,11 +24,12 @@ describe('models.Asset', () => {
   });
 
   describe('#findByUrl', ()=> {
+    beforeEach(() => Asset.findOrCreateByUrl('http://test.com'));
+
     it('should find an asset by a url', () => {
       return Asset.findByUrl('http://test.com')
         .then((asset) => {
-          expect(asset).to.have.property('url')
-            .and.to.equal('http://test.com');
+          expect(asset).to.have.property('url', 'http://test.com');
         });
     });
 

--- a/tests/models/setting.js
+++ b/tests/models/setting.js
@@ -20,20 +20,6 @@ describe('models.Setting', () => {
     });
   });
 
-  describe('#public()', () => {
-    it('should have a moderation field defined', () => {
-      return Setting.public().then(settings => {
-        expect(settings).to.have.property('moderation').and.to.equal('pre');
-      });
-    });
-
-    it('should not have the wordlist field defined', () => {
-      return Setting.public().then(settings => {
-        expect(settings).to.have.property('wordlist').and.to.have.length(0);
-      });
-    });
-  });
-
   describe('#update()', () => {
     it('should update the settings with a passed object', () => {
       const mockSettings = {moderation: 'post', infoBoxEnable: true, infoBoxContent: 'yeah'};

--- a/tests/models/user.js
+++ b/tests/models/user.js
@@ -95,12 +95,17 @@ describe('models.User', () => {
   });
 
   describe('#ban', () => {
+
     let mockComment;
+
     beforeEach(() => {
-      return Promise.all([
-        Comment.create([{body: 'testing the comment for that user if it is rejected.', id: mockUsers[0].id}])
+      return Comment.create([
+        {
+          body: 'testing the comment for that user if it is rejected.',
+          id: mockUsers[0].id
+        }
       ])
-      .then((comment) => {
+      .then(([comment]) => {
         mockComment = comment;
       });
     });
@@ -109,11 +114,11 @@ describe('models.User', () => {
       return User
         .setStatus(mockUsers[0].id, 'banned', mockComment.id)
         .then(() => {
-          User.findById(mockUsers[0].id)
-          .then((user) => {
-            expect(user).to.have.property('status')
-              .and.to.equal('banned');
-          });
+          return User.findById(mockUsers[0].id);
+        })
+        .then((user) => {
+          expect(user).to.have.property('status')
+            .and.to.equal('banned');
         });
     });
 
@@ -121,23 +126,20 @@ describe('models.User', () => {
       return User
         .setStatus(mockUsers[0].id, 'banned', mockComment.id)
         .then(() => {
-          Comment.findById(mockComment.id)
-          .then((comment) => {
-            expect(comment).to.have.property('status')
-              .and.to.equal('rejected');
-          });
+          return Comment.findById(mockComment.id);
+        })
+        .then((comment) => {
+          expect(comment).to.have.property('status')
+            .and.to.equal('rejected');
         });
     });
 
     it('should still disable and ban the user if there is no comment', () => {
       return User
         .setStatus(mockUsers[0].id, 'banned', '')
-        .then(() => {
-          User.findById(mockUsers[0].id)
-          .then((user) => {
-            expect(user).to.have.property('status')
-              .and.to.equal('banned');
-          });
+        .then(() => User.findById(mockUsers[0].id))
+        .then((user) => {
+          expect(user).to.have.property('status', 'banned');
         });
     });
   });
@@ -156,12 +158,9 @@ describe('models.User', () => {
     it('should set the status to active', () => {
       return User
         .setStatus(mockUsers[0].id, 'active', mockComment.id)
-        .then(() => {
-          User.findById(mockUsers[0].id)
-          .then((user) => {
-            expect(user).to.have.property('status')
-              .and.to.equal('active');
-          });
+        .then(() => User.findById(mockUsers[0].id))
+        .then((user) => {
+          expect(user).to.have.property('status', 'active');
         });
     });
   });

